### ISSUE TITLE
Re-organize User Preferences

### ISSUE THREAD

**Issue by horaceyoung** [feditext/feditext#89]
**Is your feature request related to a problem? Please describe.**
Currently, all user preference options are categorized into two big categories - the user preference section and the app preference section. These two categories are too broad to be meaningful.

Preference options are cluttered together without logical subsections or concise explanations of what they affect.

**Describe the solution you'd like**
Restructure the preference options into more concise, meaningful subsections with explanations where necessary. Each subsection can have flattened preference options or a NavigationView to nest the options (similar to the implementation of IceCubes).

A proposed design is listed:
- MetaSettings - Use Preferences from Server
- App Settings
    - Default Language
    - Keyboard
    - Open Links in (Default Browser / Other Apps when Available)
    - Home Timeline Position on Startup
- Appearances
    - App Icon
    - Theme
- Restrictions
    - Filters
    - Muted Users
    - Blocked Users
    - Blocked Domains
- Notifications
    - Notification Types
- Post/Toot Settings
    - Status Word
    - Default Visibility
    - Mark Content Sensitive by Default
    - Expand Media
    - Always Expand Content Warnings
    - Posting Languages
    - Show Boost and Edit Counts
    - Require Double Tap to Reblog
    - Require Double Tap to favorite
    - Hide "Show Less" Content Warning Button
    - Fold Long Text
- Media Settings
    - Autoplay GIFs
    - Autoplay Videos
    - Autoplay Animated Avatars
    - Autoplay Animated Custom Emojis
    - Autoplay Animanetd Profile Headers
    - Display Avatar Shape

**Comment by VyrCossont** [comment]
The distinction between user and app prefs is that user prefs correspond to server-side prefs in Mastodon and app prefs don't. It's not arbitrary. We can do this but it's going to mean mixing a bunch of stuff that currently corresponds relatively cleanly to that split.

**Comment by VyrCossont** [comment]
(It'd be nice to have iCloud sync for app prefs so users can have the same Feditext settings on their phones, iPads, and eventually Macs, but that's a different bug.)

**Comment by horaceyoung** [comment]
> The distinction between user and app prefs is that user prefs correspond to server-side prefs in Mastodon and app prefs don't. It's not arbitrary. We can do this but it's going to mean mixing a bunch of stuff that currently corresponds relatively cleanly to that split.

I agree currently the cut between app and user prefs is clean. Still, as pref options grow (most likely in the app prefs section) it could be hard to quickly identify the options that the user wants to change.

Would it look reasonable to you to keep user prefs as is and add more sub-sections in app prefs?

**Comment by VyrCossont** [comment]
I think we're going to need to do this after all. We're about to grow another category for accessibility.
